yt_dlp/extractor/matchtv.py

```diff
@@ -1,48 +1,28 @@
-import random
-
 from .common import InfoExtractor
-from ..utils import xpath_text
 
 
 class MatchTVIE(InfoExtractor):
-    _VALID_URL = r'https?://matchtv\.ru(?:/on-air|/?#live-player)'
+    _VALID_URL = r'https?://matchtv\.ru/on-air'
     _TESTS = [{
-        'url': 'http://matchtv.ru/#live-player',
+        'url': 'http://matchtv.ru/on-air/',
         'info_dict': {
             'id': 'matchtv-live',
-            'ext': 'flv',
+            'ext': 'mp4',
             'title': r're:^Матч ТВ - Прямой эфир \d{4}-\d{2}-\d{2} \d{2}:\d{2}$',
             'is_live': True,
         },
         'params': {
             'skip_download': True,
         },
-    }, {
-        'url': 'http://matchtv.ru/on-air/',
-        'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = 'matchtv-live'
-        video_url = self._download_json(
-            'http://player.matchtv.ntvplus.tv/player/smil', video_id,
-            query={
-                'ts': '',
-                'quality': 'SD',
-                'contentId': '561d2c0df7159b37178b4567',
-                'sign': '',
-                'includeHighlights': '0',
-                'userId': '',
-                'sessionId': random.randint(1, 1000000000),
-                'contentType': 'channel',
-                'timeShift': '0',
-                'platform': 'portal',
-            },
-            headers={
-                'Referer': 'http://player.matchtv.ntvplus.tv/embed-player/NTVEmbedPlayer.swf',
-            })['data']['videoUrl']
-        f4m_url = xpath_text(self._download_xml(video_url, video_id), './to')
-        formats = self._extract_f4m_formats(f4m_url, video_id)
+        webpage_url = 'https://video.matchtv.ru/iframe/channel/106'
+        webpage = self._download_webpage(webpage_url, video_id)
```

Comment:
should we also match this URL in the `_VALID_URL`?

Reply:
I think no

Reply:
Why do you think this should not be added?
We could just do this:
```python
    _VALID_URL = [
        r'https?://matchtv\.ru/on-air',
        r'https?://video\.matchtv\.ru/iframe/channel/106',
    ]
```

Reply:
because this link is only used as a frame

Reply:
if the user passes this URL as input we would want yt-dlp to handle it correctly though, no?

Reply:
why would a user do this
user will need to click on the first link and then copy the second one
looks pointless

Reply:
We are not taking away the first option. Both will work. I have seen some people share the iframe link online, so if someone was just given the iframe link and tried it with yt-dlp, it otherwise wouldn't work.

As I don't think this hurts to have, I'll just leave a suggestion for you to apply through the web interface.